### PR TITLE
Make waitForBestPeer sleep abortable

### PR DESCRIPTION
**Pull request by dapplion**
Partial https://github.com/ChainSafe/lodestar/issues/1499 fix. @mpetrunic seems that other streams and pushable are aborted on stop. Let me know if I'm missing something

### CODE REVIEW

**Comment by mpetrunic** [comment]
This is here is issue:
https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar/src/sync/utils/sync.ts#L96

Also, we should be able to abort rpc requests.

File: packages/lodestar/src/util/sleep.ts [hunk 1]
```diff
@@ -1,3 +1,30 @@
-export function sleep(ms: number): Promise<void> {
-  return new Promise((resolve) => setTimeout(resolve, ms));
+import {AbortSignal} from "abort-controller";
+import {ErrorAborted} from "./errors";
+
+/**
+ * Abortable sleep function. Cleans everything on all cases preventing leaks
+ * On abort throws ErrorAborted
+ */
+export async function sleep(ms: number, signal?: AbortSignal): Promise<void> {
```
Comment (wemeetagain) [inline packages/lodestar/src/util/sleep.ts]: Maybe this function should go in lodestar-utils ? we can use it in the validator and other places

**Comment by wemeetagain** [comment]
bump

**Comment by wemeetagain** [comment]
> ##[error]@chainsafe/lodestar: test/e2e/chain/genesis/genesis.test.ts(64,11): error TS2304: Cannot find name 'sleep'.

**Review by mpetrunic — DISMISSED** [review]

**Comment by dapplion** [comment]
Rebased to master to fix conflicts

File: packages/lodestar/src/sync/utils/sync.ts [hunk 2]
```diff
@@ -85,7 +86,8 @@ export function fetchBlockChunks(
   reqResp: IReqResp,
   getPeers: () => Promise<PeerId[]>,
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  maxBlocksPerChunk?: number
+  maxBlocksPerChunk?: number,
```
Comment (mpetrunic) [inline packages/lodestar/src/sync/utils/sync.ts:89]: I think you can remove this, we don't even use this in method

Comment (dapplion) [reply packages/lodestar/src/sync/utils/sync.ts:89]: Better to do it in another PR to not mixup changes?

**Review by mpetrunic — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**